cores/esp32/USBCDC.cpp

```diff
@@ -133,7 +136,8 @@ void USBCDC::begin(unsigned long baud)
     if(tx_lock == NULL) {
         tx_lock = xSemaphoreCreateMutex();
     }
-    setRxBufferSize(256);//default if not preset
+    // if rx_queue was set before begin(), keep it
+    if (!rx_queue) setRxBufferSize(256); //default if not preset
```

Comment:
The point is this function to only be called before begin and never after. If you really want to make it changeable, you need to make sure that the old queue is either empty or you copy the data from it to the new one before deletion.

Reply:
OK, I agree that in that case it shall copy data from the old queue to the new one. I'll implement it this way.

Reply:
I'll change it back to allow ```setRxBufferSize()``` to be called only before ```begin()```. This way it will be consistent with the equivalent behavior of this function in SerialHardware, as well as, to avoid issues when reducing the size of the queue (copying a bigger remaining data queue that is bigger than its new version).

Reply:
Done.